pylint/checkers/variables.py

```diff
@@ -370,6 +399,30 @@ def _ignored_modules(self):
         return get_global_option(self, 'ignored-modules', default=[])
 
 
+    @check_messages('reused-outer-loop-variable')
+    def visit_for(self, node):
+        assigned_to = _get_for_loop_assign_names(node.target)
+
+        # Only check variables that are used
+        dummy_rgx = self.config.dummy_variables_rgx
+        assigned_to = [var for var in assigned_to if not dummy_rgx.match(var)]
+
+        for variable in assigned_to:
```

Comment:
Should this handle the case where the two for statements are in different scopes? Admittedly, this is an overstretched example, but for instance I can have one outer for, which could generate some data and inside each for, I can define a new scope, such as a function or a class definition, where I can have a similar for to the outer most one, but used for a different purpose. As an example:

```
    class A(int): pass

    class B(dict): pass

    for cls in (A, B):

        class dynamic(cls):
            def test(self):
                for cls in self.__class__.mro():
                    print(cls)

        dynamic().test()
```
In this case, the check will emit a warning, no? 
What do you think? Would  it complicate too much the logic of the check?